src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs

```diff
@@ -208,10 +216,15 @@ private long GetOptions(SslProtocols protocols)
                 }
                 else
                 {
-                    inputHandle = GCHandle.Alloc(inputBuffer.token, GCHandleType.Pinned);
-                    inputPtr = Marshal.UnsafeAddrOfPinnedArrayElement(inputBuffer.token, inputBuffer.offset);
-                    done = Interop.OpenSsl.DoSslHandshake(context.DangerousGetHandle(), inputPtr, inputBuffer.size, out outputPtr, out outputSize);
-                }           
+                    unsafe
+                    {
+                        fixed (byte* tokenPtr = inputBuffer.token)
+                        {
+                            inputPtr = new IntPtr(tokenPtr + inputBuffer.offset);
+                            done = Interop.OpenSsl.DoSslHandshake(context.DangerousGetHandle(), inputPtr, inputBuffer.size, out outputPtr, out outputSize);
```

Comment:
we are doing  context.DangerousAddRef(ref gotContextReference); before calling DangerousGetHandle(). & releasing in Final. What is the issue we are seeing with that?